### ISSUE TITLE
InRange problem on Android

### ISSUE THREAD

**Issue by binaryshamy** [opencv/opencv#9837]
- OpenCV => 3.3 (and 3.2, had this then switched to 3.3)
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2017

Hello,

I have an application the runs on Windows, iOS and Android.
One of the algorithms uses inRange as one of the steps and it seems to give different results on Android compared to the other operating systems. I am using the algorithm code as a Shared project, so it's is the same on all platforms.

Initially I used this:
`cv::inRange(originalImage, 
                    cv::Scalar(colorVal.val[0] - COLOR_RANGE_B, colorVal.val[1] - COLOR_RANGE_G, colorVal.val[2] - COLOR_RANGE_R),
		    cv::Scalar(colorVal.val[0] + COLOR_RANGE_B, colorVal.val[1] + COLOR_RANGE_G, colorVal.val[2] + COLOR_RANGE_R), 
                    image);`

This returned the correct image on Windows and iOS, but on Android it gave an empty image.

Then I decided to do an inRange on each channel then combine them (the difference between results was not that big), but on Android it gave me bad results again.
The results on the Green and Blue channels were inverted (a bitwise_not on the results would bring them to be like the Windows results) but the Red channel was completely wrong and there was nothing I could do with it.

The image was grayscale on 3 channels (red, green and blue were identical).

I'm thinking this is a bug with inRange on Android but I might be wrong, so if I am doing something wrong or you have other questions please tell me.

**Comment by binaryshamy** [comment]
After analyzing the code more on Android I realized there is another problem.
It turns out that when I extracted the pixel value, used for inRange, I was getting different values from Android and Windows. In order to solve this I turned this:

`colorVal = image.at<cv::Vec3b>(y, x);`
to
`cv::Mat val = image.row(y).col(x);
colorVal = val.at<cv::Vec3b>(0,0);`

where: y = row and x = col.

After this I tried inRange on multi-channel like in the post above, but I was still getting an empty image.
But the second idea I had worked (the one described in the post above with the inRange on each channel).

In conclusion: there seems to be a problem with the method "at<>()" on Android and with the method "inRange()" on multiple channels on Android.

**Comment by adam-blip** [comment]
got the same problem, opencv inRange on android is bugged somehow
